README.md

```diff
@@ -349,6 +351,37 @@ export PINECONE_ENV="<YOUR_PINECONE_REGION>" # e.g: "us-east4-gcp"
 export MEMORY_BACKEND="pinecone"
 ```
 
+### MyScale Setup
+
+MyScale is a cloud-based database optimized for AI applications and solutions, and is a good option to store massive amount of memory with both good recall performance and efficiency.
+
+To use MyScale as AutoGPT memory backend:
+
+1. Go to [MyScale](https://myscale.com) and click on sign up.
+2. Go to your console to [create a cluster](https://docs.myscale.com/en/cluster-management/)
+3. Get your `Connection Details` under `Action` column under cluster table.
+4. Fill out the blanks in `.env`
+
+```bash
+## MYSCALE
+# ==========================
+# MYSCALE_HOST - MyScale Endpoint URL. You can find it if you log into your console.
+# MYSCALE_PORT - MyScale Endpoint port. You can find it if you log into your console. (Default: 8443)
+# MYSCALE_USERNAME - Username to access the endpoint. 
+# MYSCALE_PASSWORD - Password to access the endpoint. (Example: passwd-****)
+# MYSCALE_DATABASE - Database name you would like to use. (Default: default)
+# MYSCALE_INDEX_TYPE - Index type for vector search. (Default: IVFFLAT)
+# ==========================
+MYSCALE_HOST=your-myscale-endpoint-host
+MYSCALE_PORT=8443
+MYSCALE_USERNAME=your-username
+MYSCALE_PASSWORD=your-password
```

Comment:
is there an api key version of this?

Reply:
Actually, it is generated from the console by SaaS so maybe you can treat this as an 'separate' version of api keys. The password is not set by users but the system. It would be ok if the privacy issue was your concern in this case.